library/bignum.c

```diff
@@ -588,7 +604,7 @@ int mbedtls_mpi_write_string( const mbedtls_mpi *X, int radix,
         if( T.s == -1 )
             T.s = 1;
 
-        MBEDTLS_MPI_CHK( mpi_write_hlp( &T, radix, &p ) );
```

Comment:
Let's safe some bytes and RAM here instead, by not using `T`. Suggestion: Document that `mpi_write_hlp()` ignores the sign and just pass `X` instead of the local copy `T`.

Reply:
That's actually wrong, because we need a copy in order to successively divide `T` by the radix.

Reply:
Probably, but maybe it's better to define `T` in `mpi_write_hlp()` instead?

Reply:
I like @RonEld suggestion